arch/arm/core/cortex_m/pm_s2ram.S

```diff
@@ -81,9 +86,10 @@ SECTION_FUNC(TEXT, arch_pm_s2ram_suspend)
 	/*
 	 * Reset the marking of suspend to RAM, return is ignored.
 	 */
-	push	{r0}
 	bl pm_s2ram_mark_check_and_clear
-	pop	{r0}
+
+	/* Move system_off back to r0 as return value */
+	mov	r0, r4
```

Comment:
Bug? Was a `mov r4, r0` forgotten before the `bl pm_s2ram_mark_check_and_clear`?

This clearly changes the function's behavior: before commit, we are forwarding `system_off`'s return value, but after the commit, we're returning `system_off`, a function pointer, in place of what should be an `errno` value:
https://github.com/zephyrproject-rtos/zephyr/blob/7b4e7ff9eda0227762666a39636b5f7729fdfd51/include/zephyr/arch/common/pm_s2ram.h#L54-L56

And the following *strongly* implies that `arch_pm_s2ram_suspend` should be merely forwarding `system_off`'s return value, although I don't see it *explicitly* stated:
https://github.com/zephyrproject-rtos/zephyr/blob/7b4e7ff9eda0227762666a39636b5f7729fdfd51/include/zephyr/arch/common/pm_s2ram.h#L25-L39

Reply:
Ah, that is correct.

`r0` needs to be saved (line 32) so that `bl pm_s2ram_mark_set` on line 73 does not clobber it.
The return value of `system_off` (called on line 79 with `bx r4`) indeed needs to be saved again before calling `bl pm_s2ram_mark_check_and_clear` on line 89 (since the comment indicate that its return ought to be ignored).

Is my understanding correct that when `arch_pm_s2ram_resume` returns, it is seen as the return of `arch_pm_s2ram_suspend` when exiting a suspend?
If so, its return value should also be set to a known value even when `pm_s2ram_mark_check_and_clear` is not `1`.

Reply:
I think it's fine to overwrite `r4` since we won't need `system_off` anymore(?)

Reply:
@HesselM any update on this?

Reply:
Fixed the issue with PR #80642